src/sds.c

```diff
@@ -370,17 +339,26 @@ sds sdsResize(sds s, size_t size) {
      * to do the copy only if really needed. Otherwise if the change is
      * huge, we manually reallocate the string to use the different header
      * type. */
-    if (oldtype==type || (type < oldtype && type > SDS_TYPE_8)) {
-        newsh = s_realloc(sh, oldhdrlen+size+1);
-        if (newsh == NULL) return NULL;
-        s = (char*)newsh+oldhdrlen;
-    } else {
-        newsh = s_malloc(hdrlen+size+1);
-        if (newsh == NULL) return NULL;
-        memcpy((char*)newsh+hdrlen, s, len);
-        s_free(sh);
-        s = (char*)newsh+hdrlen;
-        s[-1] = type;
+    int is_realloc_op = (oldtype==type || (type < oldtype && type > SDS_TYPE_8));
+    size_t newlen = is_realloc_op ? oldhdrlen+size+1 : hdrlen+size+1;
+    int noop = 0;
+    #if defined(USE_JEMALLOC)
+        noop = (je_nallocx(newlen, 0) == zmalloc_size(sdsAllocPtr(s)));
+    #endif
+
+    if (!noop) {
```

Comment:
i'd rather use a strategy that doesn't cause indentation and as much blamelog changes.
the two ways i see are:
1. use `if(noop)` and then add `else if (is_realloc_op)` 
2. add `&& !noop` to the `if` and the `else` (making it an `else if`)